streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java

```diff
@@ -402,6 +405,34 @@ private void cleanGlobal(final String intermediateUserTopic) {
         Assert.assertEquals(0, exitCode);
     }
 
+    private void waitUntilUserTopicsGotDeleted() {
```

Comment:
That wound do IMHO. (1) assume "topic1" was not deleted. On the first check, "topic1" gets removed from expectedMissingTopics -- in the second check it does not get removed (because it's not in the set anymore) and thus the loop would exit -- even if the topic got not deleted. Thus, we need to add all topics in each loop, because we don't know if a topic was removed or not. (2)  adding the same topic several times is not a problem because we use a `Set`